include/alpaka/vec/Vec.hpp

```diff
@@ -79,6 +79,16 @@ namespace alpaka
         }
 #endif
 
+        template<
+            typename UDim,
+            typename UVal,
+            typename
+            = std::enable_if_t<UDim::value == TDim::value && std::is_nothrow_convertible_v<std::decay_t<UVal>, TVal>>>
```

Comment:
I am not sure about this. In alpaka we avoid everywhere implicit casts (if we have it it is IMO a bug).
This constructor opens implicit casts and we are able to convert 64bit values to 32bit values.
This could result in hidden issues due to silent loss of precision.

If I am wrong please correct me!

Reply:
Mhm, yes, this allows an implicit cast from `Vec<Dim3, uint64_t>` to `Vec<Dim3, uint32_t>`, in the same way that one can implicitly cast from `uint64_t` to `uint32_t`.

In genera I'm not fond of all the explicit casts that we need to use with alpaka, but if you don't like to relax this constraint, ok.

I will have to figure out how to fix the test without this.

Reply:
OK, it wasn't complicated :)

Reply:
By the way, maybe we could still add a conversion constructor, with some requirements:
  - if the destination type is larger than the source type (_e.g._ from `int32_t` to `int64_t`), there is not runtime check
  - otherwise, we explicitly check at runtime that the values can be stored in the destination type without loss of precision